source/module_esolver/esolver_ks_pw.cpp

```diff
@@ -816,14 +817,45 @@ void ESolver_KS_PW<T, Device>::after_scf(const int istep) {
                 double w1 = static_cast<double>(this->kv.wk[ik]
                                                 / GlobalC::ucell.omega);
 
-                for (int i = 0; i < this->pw_rho->nxyz; i++) {
+                for (int i = 0; i < this->pw_rho->nxyz; i++)
+                {
                     rho_band[i] += std::norm(wfcr[i]) * w1;
                 }
             }
 
+            // Symmetrize the charge density, otherwise the results are incorrect if the symmetry is on
+            std::cout << " Symmetrizing band-decomposed charge density..." << std::endl;
+            Symmetry_rho srho;
+            for (int is = 0; is < GlobalV::NSPIN; is++)
+            {
+                // Create pointers to the data structures
+                double** rho_save_pointers = new double*[GlobalV::NSPIN];
```

Comment:
it would be better to use std::vector than new and delete

Reply:
Done.